app/phone-list/phone-list.component.js

```diff
@@ -26,10 +27,29 @@ angular
           $scope.gridApi.core.clearAllFilters();
         };
 
-        $scope.toggleColumns = function () {
-          console.log('uiGridGridMenuService.getMenuItems($scope.gridApi) :>> ', uiGridGridMenuService.getMenuItems($scope.gridApi));
+        $scope.toggleMenu = function () {
+          /** 
+           * There are issues tring to initialize getMenuItems
+           * Could be a possible race condition between fetching the
+           * data and registering the api.
+           * 
+          */
+          if ($scope.menuItems && $scope.menuItems.length === 0) {
+            $scope.getMenuItems();
+          }
+
+          $scope.canViewMenu = !$scope.canViewMenu;
+        }
+
+        $scope.getMenuItems = function () {
+          let menuItems = uiGridGridMenuService.getMenuItems($scope.gridApi);
+          menuItems.pop(); // The last item is undefined. This is a bug in the gridApi
+          menuItems.shift(); // We want to remove 'Clear All Filters' which is default behavior in the gridApi
+          menuItems.shift(); // We want to remove 'Columns'
```

Comment:
The API is a little janky and returns 'Clear all filters, Columns, and `undefined` despite grid option settings.

We'll keep an eye on this functionality and may harden it for other scenarios as we get to them.

![Screen Shot 2022-06-13 at 3 42 56 PM](https://user-images.githubusercontent.com/106630725/173441907-be53d097-5d80-4563-9156-56b103723e11.png)